src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -346,5 +353,33 @@ private string GetEnvironmentContentNonWindows(int agentId, string agentName, IL
 
             return builder.ToString();
         }
+
+        private async Task DumpCurrentJobEventLogs(IExecutionContext executionContext, string diagFolder, DateTime jobStartTimeUtc)
+        {
+            try
+            {
+                string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+                string arguments = $@"
+                    Get-WinEvent -ListLog * `
+                    | ForEach-Object {{ Get-WinEvent -ErrorAction SilentlyContinue -FilterHashtable @{{ LogName=$_.LogName; StartTime='{ jobStartTimeUtc.ToLocalTime() }'; EndTime='{ DateTime.Now }';}} }} `
+                    | Format-List > { diagFolder }\EventViewer-{ jobStartTimeUtc.ToString("yyyyMMdd-HHmmss") }.log";
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    await processInvoker.ExecuteAsync(
+                        workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Bin),
+                        fileName: powerShellExe,
+                        arguments: arguments,
+                        environment: null,
+                        requireExitCodeZero: false,
+                        outputEncoding: null,
+                        killProcessOnCancel: false,
+                        cancellationToken: default(CancellationToken));
+                }
+            }
+            catch
+            {
+                executionContext.Debug("Failed to dump event viewer logs. Skipping.");
```

Comment:
I think that adding information about the error will be useful

Reply:
Done